massa-protocol-worker/src/node_info.rs

```diff
@@ -0,0 +1,199 @@
+//! Only public for the protocol worker.
+//! Contains all what we know for each node we are connected or we used to
+//! know in the network.
+//!
+//! # Operations
+//! Same as for wanted/known blocks, we remember here in cache wich node asked
+//! for operations and which operations he seem to already know.
+
+use massa_models::{
+    operation::OperationIds,
+    prehash::{BuildMap, Map, Set},
+};
+use massa_models::{BlockId, EndorsementId, OperationId};
+use massa_protocol_exports::ProtocolSettings;
+use std::collections::VecDeque;
+use tokio::time::Instant;
+
+/// Information about a node we are connected to,
+/// essentially our view of its state.
+///
+/// Note: should we prune the set of known and wanted blocks during lifetime of a node connection?
```

Comment:
wanted should not diverge so it's not necessary to prune it, but known must be pruned

Reply:
Related to another issue I think I will ask @adrien-zinger to open an issue on that.

Reply:
As seen with Adrien this comment is outdated.

Reply:
> "wanted"

are we talking about the blocks the node wants ? in that case it needs to be pruned. If it's the blocks we want from that node, then it's OK.

But that is orthogonal to this issue and should be handled separately. @AureliaDolo  feel free to create an issue, and resolve this comment

Reply:
He wanted to speak about operations. I close it.